utils/deploy.ts

```diff
@@ -70,29 +96,15 @@ export const writeContractsConfig = ({
   contracts: Contracts;
 }): void => {
   hre.trace('Writing contracts.json config', hre.network.name);
-  writeJsonSync(
-    path.join(__dirname, '../contracts.json'),
-    {
-      ...contractsConfig,
-      [hre.network.name]: {
-        ...contractsConfig[hre.network.name],
-        ...Object.fromEntries(
-          Object.entries(contracts)
-            .filter((_, value) => value !== undefined)
-            .map(([name, contract]) => {
-              const proxyAddress = contract?.address;
-              if (proxyAddress) {
-                // TODO: Not correct for non upgraeable
-                return [name, { proxyAddress }];
-              }
-              return [];
-            })
-        ),
-      },
-    },
-    { spaces: 2 }
-  );
-  hre.trace('Wrote contracts.json config');
+  Object.entries(contracts)
+    .filter((_, value) => value !== undefined)
+    .forEach(([name, contract]) => {
+      updateContractsConfig({
```

Comment:
in the previous version this was only opening and writing to the file once. This version does that for every contract. Any reason for the change?

Reply:
This method only ever gets called on one contract at a time now due to splitting out the deployments of the individual contracts.  This was where I stopped pulling on that thread but I could keep refactoring to make this method singular.